source/blog/036-rfc-embedded-protocol-2.md

```diff
@@ -0,0 +1,76 @@
+---
+title: "Request for Comments: New Embedded Protocol"
+author: Natalie Weizenbaum
+date: 2023-05-19 16:00 PST
+---
+
+If you're not an author of a host package for the Embedded Sass Protocol, you
+can skip this blog post—although if you're a big enough nerd, you may find it
+interesting regardless!
+
+We're planning to make a number of breaking changes to the Embedded Sass
+Protocol, and we want your feedback before we lock in the new way of doing
+things. We intend to make a number of breaking changes all at once to keep the
+total number of disruptions to a minimum.
+
+We're planning two major breaking changes:
+
+1. The Dart Sass embedded host will no longer be released as a separate
+   executable. It will now be bundled into the main Dart Sass executable,
+   accessible by running `sass --embedded`.
+
+2. Every packet in the embedded protocol now includes a compilation ID as part
+   of the packet structure, rather than declaring it in the protocol buffer
+   definitions.
+
+We're using this opportunity to also introduce three much smaller breaking
+changes:
+
+1. The specification for the embedded protocol and the protocol buffer
+   definition have been moved to [the Sass language repository] so that they can
+   be updated at the same time as changes to the language and the JS API.
+
+   [the Sass language repository]: https://github.com/sass/sass/blob/main/spec/embedded-protocol.md
+
+2. The embedded protocol now explicitly declares fields as `optional` using the
+   protocol buffers language feature. This means that "default values" for
+   various fields are no longer considered to be unset.
+
+3. The `CompilationSuccess.loaded_urls` field has been moved to
+   `CompilationResult.loaded_urls` so that it's available even when a
+   compilation fails. This allows watcher implementations to know which files to
+   watch to redo a failed compilation.
+
+The repository-organization changes have already been made, but the changes to
+the protocol itself are fully documented [in a proposal in the language
+repository].
+
+[in a proposal in the language repository]: https://github.com/sass/sass/blob/main/proposal/embedded-protocol-2.md
+
+## Combining Executables
+
+The primary benefit of folding Embedded Dart Sass into the main Dart Sass
+executable is to provide embedded hosts an easy way to expose the standard Dart
+Sass command-line API to users. Now every user who installs any embedded host
+will have the full Dart Sass executable available to them at native Dart VM
+speeds.
+
+This also helps simplify the Sass team's organization by reducing the number of
+separate repositories and release processes we need to manage.
+
+## Wire-Level Compilation ID
+
+We're pulling the compilation ID out to the protocol level in order to provide
+better parallelism, particularly on the side of the embedded compiler. Sass
```

Comment:
Nit: Should this say "concurrency" rather than "parallelism"? (if so same on line 75)

Because the change makes concurrency easier, whether it's parallel or not is up to the users, no?

I may be misremembering, but IIRC concurrency is more correct here, or did I get them the other way around? 😆